apps/OboeTester/app/src/main/java/com/google/sample/oboe/manualtest/AutomatedGlitchActivity.java

```diff
@@ -0,0 +1,111 @@
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Spinner;
+
+public class AutomatedGlitchActivity  extends BaseAutoGlitchActivity {
+
+    private Spinner mDurationSpinner;
+
+    // Test with these configurations.
+    private static final int[] PERFORMANCE_MODES = {
+            StreamConfiguration.PERFORMANCE_MODE_LOW_LATENCY,
+            StreamConfiguration.PERFORMANCE_MODE_NONE
+    };
+    private static final int[] SAMPLE_RATES = { 48000, 44100, 16000 };
+
+    private class DurationSpinnerListener implements android.widget.AdapterView.OnItemSelectedListener {
+        @Override
+        public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
+            String text = parent.getItemAtPosition(pos).toString();
+            mDurationSeconds = Integer.parseInt(text);
+        }
+
+        @Override
+        public void onNothingSelected(AdapterView<?> parent) {
+            mDurationSeconds = DEFAULT_DURATION_SECONDS;
+        }
+    }
+
+    @Override
+    protected void inflateActivity() {
+        setContentView(R.layout.activity_auto_glitches);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mDurationSpinner = (Spinner) findViewById(R.id.spinner_glitch_duration);
+        mDurationSpinner.setOnItemSelectedListener(new DurationSpinnerListener());
+    }
+
+    @Override
+    public String getTestName() {
+        return "AutoGlitch";
+    }
+
+    private void testConfiguration(int perfMode,
+                                   int sharingMode,
+                                   int sampleRate,
+                                   int inChannels,
+                                   int outChannels) throws InterruptedException {
+
+        // Configure settings
+        StreamConfiguration requestedInConfig = mAudioInputTester.requestedConfiguration;
+        StreamConfiguration requestedOutConfig = mAudioOutTester.requestedConfiguration;
+
+        requestedInConfig.reset();
+        requestedOutConfig.reset();
+
+        requestedInConfig.setPerformanceMode(perfMode);
+        requestedOutConfig.setPerformanceMode(perfMode);
+
+        requestedInConfig.setSharingMode(sharingMode);
+        requestedOutConfig.setSharingMode(sharingMode);
+
+        requestedInConfig.setSampleRate(sampleRate);
+        requestedOutConfig.setSampleRate(sampleRate);
+
+        requestedInConfig.setChannelCount(inChannels);
+        requestedOutConfig.setChannelCount(outChannels);
+
+        setTolerance(0.3f); // FIXME remove
+
+        testConfigurations();
+    }
+
+    private void testConfiguration(int performanceMode,
+                                   int sharingMode,
+                                   int sampleRate) throws InterruptedException {
+        testConfiguration(performanceMode,
+                sharingMode,
+                sampleRate, 1, 2);
```

Comment:
Magic numbers. Could you use `StreamConfiguration.PERFORMANCE_MODE_LOW_LATENCY` and `SampleRate.16000Hz`?

Reply:
The 1 and 2 are channels. I could use MONO and STEREO to be more clear. Thanks.

Reply:
Done